src/ensembl/src/shared/components/communication-framework/contact-us/contact-us-form/submit-slider/SubmitSlider.tsx

```diff
@@ -0,0 +1,225 @@
+    to: {
+      transform: 'translateX(0)'
+    }
+  }));
+
+  useEffect(() => {
+    return () => {
+      if (pressRef.current) {
+        // shouldn't happen, but being cautious
+        removeMovementListeners();
+      }
+    };
+  }, []);
+
+  useEffect(() => {
+    if (params.isDisabled) {
+      return;
+    }
+
+    params.sliderRef.current?.addEventListener('mousedown', handlePress);
+    params.sliderRef.current?.addEventListener('touchstart', handlePress);
+
+    return () => {
+      params.sliderRef.current?.removeEventListener('mousedown', handlePress);
+      params.sliderRef.current?.removeEventListener('touchstart', handlePress);
+    };
+  }, [params.isDisabled, params.sliderRef.current]);
+
+  const addMovementListeners = useCallback(() => {
+    window.addEventListener('mousemove', handleDrag);
+    window.addEventListener('touchmove', handleDrag);
+    window.addEventListener('mouseup', handleRelease);
+    window.addEventListener('touchend', handleRelease);
+  }, []);
+
+  const removeMovementListeners = useCallback(() => {
+    window.removeEventListener('mousemove', handleDrag);
+    window.removeEventListener('touchmove', handleDrag);
+    window.removeEventListener('mouseup', handleRelease);
+    window.removeEventListener('touchend', handleRelease);
+  }, []);
+
+  const handlePress = useCallback((event: TouchEvent | MouseEvent) => {
+    pressRef.current = true;
+    recordInitialPointerX(event);
+    recordSliderRects();
+    setGrabbingCursor(true);
+    addMovementListeners();
+  }, []);
+
+  const handleRelease = useCallback(() => {
+    pressRef.current = false;
+    updateTranslateX(0);
+    setGrabbingCursor(false);
+    removeMovementListeners();
+    params.onRelease();
+  }, []);
+
+  const handleDrag = useCallback((event: TouchEvent | MouseEvent) => {
+    const distance = getDistance(event);
+
+    if (hasNotReachedTrackEnd(distance)) {
+      updateTranslateX(distance);
+    } else {
+      handleRelease();
+      params.onSlideCompleted();
+    }
+  }, []);
+
+  const setGrabbingCursor = (isGrabbing: boolean) => {
+    if (isGrabbing) {
+      document.body.style.cursor = 'grabbing';
+    } else {
+      document.body.style.cursor = '';
+    }
+  };
+
+  const recordSliderRects = () => {
+    const trackRect =
+      params.trackRef.current?.getBoundingClientRect() as DOMRect;
+    const sliderRect =
+      params.sliderRef.current?.getBoundingClientRect() as DOMRect;
+    sliderRectsRef.current = { trackRect, sliderRect };
+  };
+
+  const recordInitialPointerX = (event: TouchEvent | MouseEvent) => {
+    initialPointerX.current = getCurrentPointerX(event);
+  };
+
+  const getDistance = (event: TouchEvent | MouseEvent) => {
+    const currentPointerX = getCurrentPointerX(event);
+    const trackRect = sliderRectsRef.current?.trackRect as DOMRect;
+    const sliderRect = sliderRectsRef.current?.sliderRect as DOMRect;
+    const minValue = 0;
+    const maxValue = trackRect.width - sliderRect.width;
```

Comment:
Should we really check if the user have dragged to the far end?  Isn't it better to automatically complete the slide once the user drag reaches, say 75% for a better user experience?

Reply:
Decided that a slide all the way to the right margin is good for now; will hear from Andrea and Lucy if they want it to be changed.